### PR TITLE
Update rollup dependencies and make watcher (tests) a little more stable

### PR DESCRIPTION
**Pull request by lukastaegert**
As I was unnerved by the constant reminders that some of our dependencies had security issues, I finally updated everything (including transitive dependencies) to their latest versions.

After that I started to have issues with the watcher tests which lead to a prolonged debug session. There may been some overlap with @guybedford's work on the watchers, so let's see about that.

* After the update, I was having issues that the watcher was sometimes starting its next run too early before we were actually looking for watcher events. Could be some changes in "sander", not sure. I managed to fix this by increasing the wait period in the watcher to 200ms.
* The "rerun" logic sometimes seemed a little shaky and cause random issues. I tried to clean this up a bit. Most notably, reruns are now always triggered indirectly via "invalidate". Otherwise it could happen that two concurrent runs were started.

### CODE REVIEW

**Comment by lukastaegert** [comment]
Also did some tweaking to the CI script. The use of `npm ci` instead of `npm install` should trigger an error when the package-lock.json file is not in sync with the package.json file.

File: tsconfig.json [hunk 1]
```diff
@@ -11,7 +11,6 @@
         "noUnusedParameters": true,
         "pretty": true,
         "skipLibCheck": true,
-        "strictPropertyInitialization": true,
```
Comment (guybedford) [inline tsconfig.json]: Why do we need to remove this?

File: package.json [hunk 2]
```diff
@@ -61,52 +61,50 @@
...
     "rollup-plugin-json": "^3.0.0",
     "rollup-plugin-node-resolve": "^3.3.0",
     "rollup-plugin-replace": "^2.0.0",
-    "rollup-plugin-string": "^2.0.0",
+    "rollup-plugin-string": "^2.0.2",
     "rollup-plugin-typescript": "^0.8.1",
-    "rollup-pluginutils": "^2.2.0",
-    "rollup-watch": "^4.3.1",
+    "rollup-pluginutils": "^2.3.0",
     "sander": "^0.6.0",
```
Comment (guybedford) [inline package.json]: Not sure the hypothesis stands on the delay?

File: src/watch/index.ts [hunk 3]
```diff
@@ -60,35 +60,30 @@ export class Watcher extends EventEmitter {
...
 			.then(() => {
-				return minBuildDelayPromise;
-			})
-			.then(() => {
-				this.running = false;
-
 				if (this.rerun) {
 					this.rerun = false;
-					this.run();
+					this.invalidate();
```
Comment (guybedford) [inline src/watch/index.ts]: This definitely handles those edge cases much more nicely.

File: package.json (hunk 2 above)
Comment (lukastaegert) [reply package.json]: What I observed is that the watcher would trigger noticeably before sander.writeFileSync would return. Might also be related to an update of a transitive dependency.

**Pull request merged**